site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+
+## Merge-on-read
+
+In the next few sections, we provide more details around the Iceberg MoR design.
+
+### Row-Level Delete File Spec
+
+As documented in the [Spec](../spec/#row-level-deletes) page, Iceberg supports 2 different types of row-level delete files: **position deletes** and **equality deletes**.
+If you are unfamiliar with these concepts, please read the related sections in the spec for more information before proceeding.
+
+Also note that because row-level delete files are valid Iceberg data files, each file must define the partition it belongs to.
+If the file belongs to `Unpartitioned` (the partition spec has no partition field), then the delete file is called a **global delete**. 
+Otherwise, it is called a **partition delete**.
+
+### MoR Update as Delete + Insert
+
+In Iceberg, update is modeled as a delete with an insert within the same transaction, so there is no concept of an "update file".
+During a MoR write transaction, new data files and delete files are committed with the same sequence number.
+During a MoR read process, delete files are applied to data files of strictly lower sequence numbers.
+This ensures the latest updated rows are displayed to users during a MoR read.
+
+### Delete File Writer
+
+From the end user's perspective, it is very rare that they could directly request deletion of a specific row of a specific file given the abstraction provided by Iceberg. 
+A delete requirement almost always comes as a predicate such as `id = 5` or `date < '2000-01-01'`. 
+Given the predicate, a delete writer can write delete files in one or some combinations of the following ways:
+
+1. **partition position deletes**: perform a scan \[1\] to know the data files and row positions affected by the predicate and then write partition \[2\] position deletes
+2. **partition equality deletes**: convert input predicate \[3\] to partition equality predicates and write partition equality deletes
+3. **partition global deletes**: convert input predicate to equality predicates and write global equality deletes 
+
+\[1\] scan here can mean a table scan, or a scan of unflushed files (stored in memory, local RocksDB, etc.) for use cases like streaming
+
+\[2\] it is in theory possible to write global position deletes, but the writer already knows the exact partitions to write, so it is almost always preferred to write partition position deletes because it costs the same and improves the efficiency of the MoR read process.
+
+\[3\] if the input inequality predicate cannot be converted to a finite number of equality predicates (e.g. `price > 2.33`), then it is only possible to use position deletes instead.
+
+### Data File Reader with Deletes
+
+During MoR read time, the Iceberg reader indexes all the delete files and determines the associated delete files for each data file. Typically,
+
+- as described before, delete files are only applied to data files that has a strictly lower sequence number
```

Comment:
Again, please make it consistent with the spec which says position deletes should be applied to data files with the same sequence number.